### PR TITLE
tests: refactor plugin dependency tests

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Replace redundant fixtures with a fixture factory.

### CODE REVIEW

**Review by sergiusens — COMMENTED** [review]
Looks good, but should this fixture be only one and in conftest.py?

**Comment by mr-cal** [comment]
Ah, `conftest.py` is exactly what I was looking for.  Thanks.

File: tests/conftest.py [hunk 1]
```diff
@@ -116,3 +118,38 @@ def fake_snapd():
 def fake_snap_command(mocker):
     """Mock the snap command."""
     return FakeSnapCommand(mocker)
+
+
+@pytest.fixture()
+def dependency_fixture(new_dir):
+    """Fixture factory for dependencies."""
+
+    def _dependency_fixture(
```
Comment (sergiusens) [inline tests/conftest.py:127]: since the scope of this is dependency_fixture, this does not need to be private

**Review by sergiusens — COMMENTED** [review]
Looks good, just a minor nit

**Review by sergiusens — APPROVED** [review]

**Pull request merged**